service/matching/task_queue_manager.go

```diff
@@ -809,20 +818,52 @@ func (c *taskQueueManagerImpl) RedirectToVersionedQueueForAdd(ctx context.Contex
 
 	if c.kind == enumspb.TASK_QUEUE_KIND_STICKY {
 		// In the sticky case we don't redirect, but we may kick off this worker if there's a newer one.
-		err := checkVersionForStickyAdd(data, buildId)
-		return c.taskQueueID, userDataChanged, err
+		unknownBuild, err := checkVersionForStickyAdd(data, buildId)
+		if err != nil {
+			return nil, nil, err
+		}
+		if unknownBuild {
+			c.recordUnknownBuildTask(buildId)
+			// Don't bother persisting the unknown build id in this case: sticky tasks have a
+			// short timeout, so it doesn't matter if they get lost.
+		}
+		return c.taskQueueID, userDataChanged, nil
 	}
 
-	versionSet, err := lookupVersionSetForAdd(data, buildId)
+	versionSet, unknownBuild, err := lookupVersionSetForAdd(data, buildId)
```

Comment:
I'm worried that if there's another non-retryable error like an unknown build ID here, then the number of branches here will just explode. Is there a way we can just stuff the unknownBuild flag into the error type here and treat it the same as any other non-retryable error?

Reply:
unknownBuild being true isn't an error, though. If I return both a value and an "error", won't you complain that that's confusing not idiomatic?